test/e2e/UniswapTrade.t.sol

```diff
@@ -0,0 +1,189 @@
+            GPv2Signing.Scheme.Eip712,
+            0
+        );
+
+        // mint some dai
+        dai.mint(trader2.addr, 300.3 ether);
+        vm.prank(trader2.addr);
+        dai.approve(vaultRelayer, type(uint256).max);
+
+        // place order to buy 0.5 wETH for max 300 dai
+        encoder.signEncodeTrade(
+            vm,
+            trader2,
+            GPv2Order.Data({
+                kind: GPv2Order.KIND_BUY,
+                partiallyFillable: false,
+                sellToken: dai,
+                buyToken: wETH,
+                sellAmount: 300 ether,
+                buyAmount: 0.5 ether,
+                feeAmount: 0.3 ether,
+                validTo: 0xffffffff,
+                appData: bytes32(uint256(1)),
+                sellTokenBalance: GPv2Order.BALANCE_ERC20,
+                buyTokenBalance: GPv2Order.BALANCE_ERC20,
+                receiver: GPv2Order.RECEIVER_SAME_AS_OWNER
+            }),
+            domainSeparator,
+            GPv2Signing.Scheme.Eip712,
+            0
+        );
+
+        // interaction to swap the remainder on uniswap
+        encoder.addInteraction(
+            GPv2Interaction.Data({
+                target: address(wETH),
+                value: 0,
+                callData: abi.encodeCall(IERC20.transfer, (address(uniswapPair), uniswapWethInAmount))
+            }),
+            SettlementEncoder.InteractionStage.INTRA
+        );
+        (uint256 amount0Out, uint256 amount1Out) =
+            isWethToken0 ? (uint256(0), uniswapDaiOutAmount) : (uniswapDaiOutAmount, uint256(0));
+        encoder.addInteraction(
+            GPv2Interaction.Data({
+                target: address(uniswapPair),
+                value: 0,
+                callData: abi.encodeCall(IUniswapV2Pair.swap, (amount0Out, amount1Out, address(settlement), hex""))
+            }),
+            SettlementEncoder.InteractionStage.INTRA
+        );
+
+        // set token prices
+        IERC20[] memory tokens = new IERC20[](2);
+        tokens[0] = wETH;
+        tokens[1] = dai;
+        uint256[] memory prices = new uint256[](2);
+        prices[0] = uniswapDaiOutAmount;
+        prices[1] = uniswapWethInAmount;
+        encoder.tokenRegistry.tokenRegistry().setPrices(tokens, prices);
+
+        SettlementEncoder.EncodedSettlement memory encodedSettlement = encoder.encode(settlement);
+
+        vm.prank(solver);
+        settle(encodedSettlement);
+
+        assertEq(wETH.balanceOf(address(settlement)), 0.001 ether, "weth fees not as expected");
+        assertEq(dai.balanceOf(address(settlement)), 0.3 ether, "dai fees not as expected");
+
+        assertEq(wETH.balanceOf(trader1.addr), 0, "not all weth sold");
+        assertEq(dai.balanceOf(trader1.addr), uniswapDaiOutAmount * 2, "dai received not as expected");
+
+        assertEq(wETH.balanceOf(trader2.addr), 0.5 ether, "weth bought not correct amount");
+        assertEq(dai.balanceOf(trader2.addr), 300.3 ether - (uniswapDaiOutAmount + 0.3 ether));
+    }
```

Comment:
Is there a reason why the uniswap reserve checks post-swap haven't been asserted?